airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -561,11 +580,8 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                 try:
                     stream_slice = next(report_stream.stream_slices(sync_mode=SyncMode.full_refresh))
                     next(report_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice), None)
-                except HTTPError as e:
-                    error_response = ""
-                    if e.response.status_code == HTTPStatus.BAD_REQUEST:
-                        error_response = e.response.json().get("error", {}).get("message", "")
-                    return False, WRONG_CUSTOM_REPORT_CONFIG.format(report=report["name"], error_response=error_response)
+                except MessageRepresentationAirbyteTracedErrors as e:
+                    return False, f"{e.message}. {self._extract_message_details(e.internal_message)}"
```

Comment:
d.- get_error_mapping is implemented in **GoogleAnalyticsDataApiBaseStream**. 

Previously, check_connection in **SourceGoogleAnalyticsDataApi** would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method _get_google_analytics_data_api_base_error_mapping_ handles this task.